layer/frame_time_layer.cc

```diff
@@ -99,10 +98,7 @@ class FrameTimeLayerData : public performancelayers::LayerData {
   std::string benchmark_start_pattern_;
   std::optional<performancelayers::LogScanner> benchmark_log_scanner_;
 
-  mutable absl::Mutex queue_to_device_lock_;
-  // The map from a queue to the device that owns it.
-  absl::flat_hash_map<VkQueue, VkDevice> queue_to_device_
-      ABSL_GUARDED_BY(queue_to_device_lock_);
+  performancelayers::QueueToDeviceMap queue_to_device_map_;
```

Comment:
There's some potential to make usage more concise if we inherit this class as a mixin, instead of adding it as a member. But I think multiple inheritance is discouraged, and I'm fine with a few single-line methods to explicitly forward work the the right member.